### PR TITLE
Add option to ignore non literal constant strings

### PR DESCRIPTION
**Pull request by fbaierl**
See issue: https://github.com/xitrum-framework/scala-xgettext/issues/19

### CODE REVIEW

File: src/main/scala/scala/Xgettext.scala [hunk 1]
```diff
@@ -125,21 +131,30 @@ msgstr ""
...
               } else if (i18n_tcn.contains(methodName)) {
-                val msgctxt     = fixBackslashSingleQuote(stringConstant(list.head, pos))
-                val msgid       = fixBackslashSingleQuote(stringConstant(list(1), pos))
-                val msgidPlural = fixBackslashSingleQuote(stringConstant(list(2), pos))
-                msgToLines.addBinding((Some(msgctxt), msgid, Some(msgidPlural)), line)
+                (stringConstant(list.head, pos), stringConstant(list(1), pos), stringConstant(list(2), pos)) match {
+                  case (Some(msgctxt), Some(msgid), Some(msgidPlural)) =>
+                    msgToLines.addBinding((Some(fixBackslashSingleQuote(msgctxt)), fixBackslashSingleQuote(msgid),
+                      Some(fixBackslashSingleQuote(msgidPlural))), line)
+                  case _ => // do nothing
```
Comment (ngocdaothanh) [inline src/main/scala/scala/Xgettext.scala]: The `case _ => // do nothing` doesn't look very nice.

Let's use for comprehension, like this:

```
for (msgid <- stringConstant(list.head, pos)) {
  msgToLines.addBinding(
    (None, fixBackslashSingleQuote(msgid), None),
    line
  )
}

for (msgid <- stringConstant(list.head, pos); msgidPlural <- stringConstant(list(1), pos)) {
  msgToLines.addBinding(
    (None, fixBackslashSingleQuote(msgid), Some(fixBackslashSingleQuote(msgidPlural))),
    line
  )
}

for (msgctxt <- stringConstant(list.head, pos); msgid <- stringConstant(list(1), pos)) {
  msgToLines.addBinding(
    (Some(fixBackslashSingleQuote(msgctxt)), fixBackslashSingleQuote(msgid), None),
    line
  )
}

for (msgctxt <- stringConstant(list.head, pos); msgid <- stringConstant(list(1), pos); msgidPlural <- stringConstant(list(2), pos)) {
  msgToLines.addBinding(
    (Some(fixBackslashSingleQuote(msgctxt)), fixBackslashSingleQuote(msgid), Some(fixBackslashSingleQuote(msgidPlural))),
    line
  )
}
```

Comment (fbaierl) [reply src/main/scala/scala/Xgettext.scala]: Right, updated my PR.

File: src/main/scala/scala/Xgettext.scala [hunk 2]
```diff
@@ -125,21 +131,26 @@ msgstr ""
...
+                for (msgid <- stringConstant(list.head, pos);
+                     msgidPlural <- stringConstant(list(1), pos)) {
+                  msgToLines.addBinding((None, fixBackslashSingleQuote(msgid), Some(fixBackslashSingleQuote(msgidPlural))), line)
+                }
               } else if (i18n_tc.contains(methodName)) {
-                val msgctxt = fixBackslashSingleQuote(stringConstant(list.head, pos))
-                val msgid   = fixBackslashSingleQuote(stringConstant(list(1), pos))
-                msgToLines.addBinding((Some(msgctxt), msgid, None), line)
+                for (msgctxt <- stringConstant(list.head, pos);
+                     msgid <- stringConstant(list(1), pos)){
```
Comment (ngocdaothanh) [inline src/main/scala/scala/Xgettext.scala]: For style consistency, there should be a space between `)` and `{`. But that's OK.

**Comment by ngocdaothanh** [comment]
I've just released version 1.5.2 to Maven. You should be able to download that very soon.

**Comment by fbaierl** [comment]
Great, thank you. :)

**Pull request merged**